java/client/src/main/java/compatibility/clients/jedis/JedisPool.java

```diff
@@ -0,0 +1,314 @@
+        Jedis jedis = availableConnections.poll();
+        if (jedis != null && !jedis.isClosed()) {
+            jedis.resetForReuse();
+            activeConnections.incrementAndGet();
+            return jedis;
+        }
+
+        // Create a new connection if pool is not at capacity
+        if (totalConnections.get() < maxConnections) {
+            jedis = createNewConnection();
+            if (jedis != null) {
+                totalConnections.incrementAndGet();
+                activeConnections.incrementAndGet();
+                return jedis;
+            }
+        }
+
+        // Wait for an available connection
+        try {
+            jedis = availableConnections.poll(maxWaitMillis, TimeUnit.MILLISECONDS);
+            if (jedis != null && !jedis.isClosed()) {
+                jedis.resetForReuse();
+                activeConnections.incrementAndGet();
+                return jedis;
+            } else {
+                throw new JedisException("Could not get a resource from the pool");
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new JedisException("Interrupted while waiting for connection", e);
+        }
+    }
+
+    /**
+     * Return a connection to the pool. This method is called internally by Jedis.close() for pooled
+     * connections.
+     *
+     * @param jedis the connection to return
+     */
+    protected void returnResource(Jedis jedis) {
+        if (!closed.get() && jedis != null && !jedis.isClosed()) {
+            activeConnections.decrementAndGet();
+            availableConnections.offer(jedis);
+        } else if (jedis != null) {
+            // Connection is closed or pool is closed, decrement total count
+            totalConnections.decrementAndGet();
+            if (!jedis.isClosed()) {
+                try {
+                    jedis.getGlideClient().close();
+                } catch (Exception e) {
+                    // Log error but continue
+                    System.err.println("Error closing returned connection: " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create a new Jedis connection.
+     *
+     * @return a new Jedis connection
+     * @throws JedisException if unable to create connection
+     */
+    private Jedis createNewConnection() {
+        try {
+            GlideClientConfiguration glideConfig =
+                    ConfigurationMapper.mapToGlideConfig(host, port, config);
+            GlideClient glideClient = GlideClient.createClient(glideConfig).get();
+            return new Jedis(glideClient, this, config);
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("Failed to create new connection", e);
+        }
+    }
+
+    /** Close the pool and all connections. */
+    @Override
+    public void close() {
+        if (closed.compareAndSet(false, true)) {
```

Comment:
We should notify any threads that are waiting in getResource() for an available connection that the pool is now closed. This means we need to keep a count of waiters and push nulls to tell them they aren't getting valid Jedis objects. (OK with deferring this)